openquake/hazardlib/gsim/tusa_langer_2015.py

```diff
@@ -0,0 +1,375 @@
+        for spec of input and result values.
+        """
+        # extracting dictionary of coefficients specific to required
+        # intensity measure type 
+        C = self.COEFFS[imt]
+
+	imean = (self._compute_magnitude(rup, C) +
+                 self._compute_distance(rup, dists, C)+
+		 self._get_site_amplification(sites, C)) 
+                    	
+        istddevs = self._get_stddevs(C, stddev_types, num_sites=sites.vs30.size)
+
+        # convert from log10 to ln and from cm/s**2 to g
+        mean = np.log((10.0 ** (imean - 2.0)) / g)
+        
+        # Return stddevs in terms of natural log scaling
+        stddevs = np.log(10.0 ** np.array(istddevs))
+       
+        return mean, stddevs
+
+    def _get_stddevs(self, C, stddev_types, num_sites):
+        """
+        Return standard deviations as defined in tables below
+        """
+       	assert all(stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+                   for stddev_type in stddev_types)
+        stddevs = [np.zeros(num_sites) + C['SigmaTot'] for _ in stddev_types]
+        return stddevs
+
+    def _compute_distance(self, rup, dists, C):
+        """
+        Compute the distance function:
+        ``c1 + c2 * (M-Mref) * log(sqrt(repi ** 2 + h ** 2)/Rref) -
+             c3*(sqrt(repi ** 2 + h ** 2)-Rref)``
+        """
+        mref = 3.6
+        rref = 1.0
+        rval = np.sqrt(dists.repi ** 2 + C['h'] ** 2)
+        return (C['c1'] + C['c2'] * (rup.mag - mref)) *\
+            np.log10(rval / rref) + C['c3'] * (rval - rref)
+
+    def _compute_magnitude(self, rup, C):
+        """
+        Compute the magnitude function:
+        a + b1 * (M) + b2 * (M)**2 for M<=Mh
+         """
+        
+        return C['a'] + (C['b1'] * (rup.mag)) +\
+                (C['b2'] * (rup.mag) ** 2)
+
+    def _get_site_amplification(self, sites, C):
+	
+	"""
+	Compute the site amplification function given by FS = eiSi,
+        for i = 1,2,3 where Si are the coefficients determined through 
+        regression analysis, and ei are dummy variables (0 or 1) used 
+        to denote the different EC8 site classes.
+	"""
+
+	ssa, ssb, ssd = self._get_site_type_dummy_variables(sites)
+	
+	return (C['sA'] * ssa) + (C['sB'] * ssb) + (C['sD'] * ssd) 
+	
+    def _get_site_type_dummy_variables(self, sites):
+
+	"""
+	Get site type dummy variables, which classified the sites into different
+        site classes based on the shear wave velocity in the upper 30 m (Vs30)
+	according to the EC8 (CEN 2003):
+	class A: Vs30 > 800 m/s
+	class B: Vs30 = 360 - 800 m/s
+	class C*: Vs30 = 180 - 360 m/s
+	class D: Vs30 < 180 m/s
+	class E*: 5-20m of C- or D-type alluvium underlain by material with Vs30 > 800 m/s.
+        *Not computed by this GMPE
+	"""
+	ssa = np.zeros(len(sites.vs30))
+	ssb = np.zeros(len(sites.vs30))
+	ssd = np.zeros(len(sites.vs30))
+        
+		
+	# Class D; Vs30 < 180 m/s.
+	idx = (sites.vs30 >= 1E-10) & (sites.vs30 < 180.0) 
```

Comment:
I don't think you need the two logical commands here, just running sites.v30 < 180.0 should be enough.